src/main/java/com/datalogics/pdf/samples/manipulation/ConvertToPdfA2.java

```diff
@@ -0,0 +1,244 @@
+     *
+     * @param inputUrl The URL of the document to be converted
+     * @param outputUrl The URL of the converted document
+     * @throws IOException an I/O operation failed or was interrupted
+     * @throws PDFFontException there was an error in the font set or an individual font
+     * @throws PDFInvalidDocumentException a general problem with the PDF document, which may now be in an invalid state
+     * @throws PDFIOException there was an error reading or writing a PDF file or temporary caches
+     * @throws PDFSecurityException some general security issue occurred during the processing of the request
+     * @throws PDFInvalidParameterException one or more of the parameters passed to a method is invalid
+     * @throws PDFUnableToCompleteOperationException the operation was unable to be completed
+     */
+    public static void convertToPdfA2(final URL inputUrl, final URL outputUrl,
+                                      final PDFA2ConformanceLevel conformanceLevel)
+                    throws IOException, PDFFontException, PDFInvalidDocumentException, PDFIOException,
+                    PDFSecurityException, PDFInvalidParameterException, PDFUnableToCompleteOperationException {
+        ByteWriter writer = null;
+        // Attach font set to PDF
+        final PDFFontSet pdfaFontSet = FontSetLoader.newInstance().getFontSet();
+        final PDFOpenOptions openOptions = PDFOpenOptions.newInstance();
+        openOptions.setFontSet(pdfaFontSet);
+
+        final PDFDocument pdfDoc = DocumentUtils.openPdfDocumentWithOptions(inputUrl, openOptions);
+
+        final PDFA2ConversionOptions options = new PDFA2ConversionOptions();
+        final PdfA2ConversionHandler handler = new PdfA2ConversionHandler();
+
+        // Tell font set to embed fonts
+        options.setShouldEmbedFonts(true);
+
+        // Set an OutputIntent
+        if (options.getPDFAOutputIntent() == null) {
+            final PDFOutputIntent outputIntent = PDFOutputIntent.newInstance(pdfDoc, "GTS_PDFA1", "CGATS TR 001");
+            outputIntent.setOutputCondition("U.S. Web Coated(SWOP)v2");
+            outputIntent.setRegistryName("http://www.color.org");
+            final ICC_Profile profile = ICC_Profile.getInstance(ConvertToPdfA2.class.getResourceAsStream(SRGB_PROFILE));
+            final PDFICCProfile destProfile = PDFICCProfile.newInstance(pdfDoc, profile);
+            outputIntent.setDestOutputProfile(destProfile);
+            outputIntent.setOutputConditionIdentifier("CGATS TR 001");
+            final PDFOutputIntentsList outputIntentsList = PDFOutputIntentsList.newInstance(pdfDoc);
+            outputIntentsList.add(outputIntent);
+            pdfDoc.requireCatalog().setOutputIntents(outputIntentsList);
+            handler.pdfaOutputIntentSet(outputIntent);
+        }
+
+        // Provide ICC profiles for default color spaces
+        final InputStream rgbInput = ConvertToPdfA2.class.getResourceAsStream(RGB_PROFILE);
```

Comment:
Turns out the `PDFAConversionOptionsFactory` does this too, see [here](https://octocat.dlogics.com/datalogics/pdf-java-toolkit/blob/develop/pdfjt/src/main/java/com/adobe/pdfjt/services/pdfa/PDFAConversionOptionsFactory.java#L142).